### PR TITLE
Create a separate test for each layout configuration in HomePageConfigurationTests

### PR DESCRIPTION
**Pull request by mmatoszko**
Task/Issue URL:
Tech Design URL:
CC:

**Description**:
Main reasons for this change:
- Having a separate test case for each test configuration improves the readability
- Using a dedicated 'assertLayout' function with '#file' and '#line' macro parameters show test failures nicely in the Xcode
- The test cases were doubled, the duplicates were removed

**Steps to test this PR**:
1. code review

**Orientation Testing**:
Not needed

**Device Testing**:
Not needed

**OS Testing**:
Not needed

**Theme Testing**:
Not needed
---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGoTests/HomePageConfigurationTests.swift [hunk 1]
```diff
@@ -24,81 +24,68 @@ import XCTest
...
 class HomePageConfigurationTests: XCTestCase {
 
-    struct Test {
-
-        let layout: HomePageLayout
-        let favorites: Bool
-        let links: [Link]
-        let expected: [HomePageConfiguration.Component]
-
+    func testLayoutCentered() {
```
Comment (bwaresiak) [inline DuckDuckGoTests/HomePageConfigurationTests.swift]: Convention that we try to use for test names is `testWhen<condition>Then<expectation>`, e.g. `testWhenCenteredAndFavoritesDisabledThenFixedCenteredIsUsed`

File: DuckDuckGoTests/HomePageConfigurationTests.swift [hunk 2]
```diff
@@ -24,81 +24,68 @@ import XCTest
...
+        let test = Test(layout: .centered, favorites: false, links: [])
+        assertLayout(test: test, expected: [ .centeredSearch(fixed: true) ])
     }
 
-    func test() {
+    func testLayoutCenteredWithFavourites() {
```
Comment (bwaresiak) [inline DuckDuckGoTests/HomePageConfigurationTests.swift]: We are sticking to American english, so `Favorite` :)

**Review by bwaresiak — CHANGES_REQUESTED** [review]
Hello,

Sorry it took us a while to pick this up, I think what you've proposed makes sense and increases clarity of these tests. I've added some comments inline.

Thanks for contribution. :)

Internal reference: https://app.asana.com/0/856498667320406/1184665589180937

**Comment by mmatoszko** [comment]
@bwaresiak thanks for your suggestions! Should be good now

**Review by bwaresiak — APPROVED** [review]
Yup, all good! Thanks :)

**Pull request merged**